### PR TITLE
Update with new year (2024)

### PR DESCRIPTION
**Pull request by SutuSebastian**
- [x] I have followed the [Your First Code Contribution section of the Contributing guide](https://github.com/themesberg/flowbite-react/blob/main/CONTRIBUTING.md#your-first-code-contribution)

New year, new resolution 😎

### Changes

- [x] make footer year dynamic
- [x] manually update `license.mdx` year
- [x] fix `Datepicker` manual year tests

### Result

Footer
Screenshot 2024-01-05 at 09 50 07

### CODE REVIEW

File: content/docs/getting-started/license.mdx [hunk 1]
```diff
@@ -5,7 +5,7 @@ description: Learn more about the open-source licensing rights of the Flowbite R
 
 ## MIT License
 
-Copyright &copy; 2023 Bergside Inc.
+Copyright &copy; 2024 Bergside Inc.
```
Comment (rluders) [inline content/docs/getting-started/license.mdx:8]: Stupid question... Can't it be dynamic?

**Review by rluders — CHANGES_REQUESTED** [review]

File: content/docs/getting-started/license.mdx (hunk 1 above)
Comment (SutuSebastian) [reply content/docs/getting-started/license.mdx:8]: Not stupid at all, I already thought of this and it seemed weird to put a JS snippet there for the readers that open it in github or any `mdx` viewer.

Comment (SutuSebastian) [reply content/docs/getting-started/license.mdx:8]: Just checked some other big repos and it seems like not even `tailwindcss` includes the hardcoded year 🤔

Comment (rluders) [reply content/docs/getting-started/license.mdx:8]: Well, it should be possible to make it dynamic somehow. I can't imagine creating a PR every single year just to update the year at the footer. 🤣

Comment (SutuSebastian) [reply content/docs/getting-started/license.mdx:8]: so footer problem is gone, we're talking here only `license.mdx` file 😄 which is in docs

Comment (rluders) [reply content/docs/getting-started/license.mdx:8]: Yeah, got confused by the image. LOL

**Review by tulup-conner — APPROVED** [review]

**Pull request merged**